XenAdmin/Wizards/NewNetworkWizard.cs

```diff
@@ -227,7 +227,17 @@ private void CreateSRIOV()
         private void CreateNonBonded()
         {
             XenAPI.Network network = PopulateNewNetworkObj();
-            PIF nic = pageNetworkDetails.SelectedHostNic;
+
+            PIF nic;
+            if(pageNetworkDetails.createVlanOnSriovNetwork)
+            {
+                var sriovLogicalPif = pageNetworkDetails.SelectedHostNic.sriov_physical_PIF_of[0];

```

Comment:
To be on the safe side, we should check that pageNetworkDetails.SelectedHostNic.sriov_physical_PIF_of.Count > 0